packages/app/src/lib/workers/backendWorker.ts

```diff
@@ -769,3 +795,106 @@ class StreamMaterializer {
     });
   }
 }
+
+class PreviewMaterializer {
```

Comment:
Just a note, this should be unnecessary as soon as we get partial space loading.

Then we can just open a non-joined space like we would any other space and it will load only the events it needs.